src/Command/CodeStudio/CodeStudioSetCICDVarCommand.php

```diff
@@ -0,0 +1,213 @@
+    // @todo: Check if there is something for environment variable permission.
+    $allowedProjects = $this->gitLabClient->projects()->all(['owned' => TRUE]);
+
+    // If there are no project available for the user.
+    if (count($allowedProjects) === 0) {
+      $this->io->error('There are no code studio projects available for your account.');
+      return 1;
+    }
+
+    $list = [];
+    foreach ($allowedProjects as $allowedProject) {
+      $list[$allowedProject['id']] = $allowedProject['name'];
+    }
+
+    $project_id = NULL;
+    if ($input->getOption('gitlab-project-id')) {
+      $project_id = $this->input->getOption('gitlab-project-id');
+      // If project id provided by user is not valid.
+      if (!isset($list[$project_id])) {
+        $this->io->error('Project id provided is not valid. Please check the project id.');
+        return 1;
+      }
+    }
+
+    // If user not providing project id in option.
+    if (!$project_id) {
+      // Prepare list of projects to ask from the user.
+      $project_choice_labels = array_values($list);
+      $question = new ChoiceQuestion('Please select a Cloud Platform subscription',
+        $project_choice_labels,
+        $project_choice_labels[0],
+      );
+      $choice_id = $this->io->askQuestion($question);
+      $project_id = array_search($choice_id, $list, TRUE);
+    }
+
+    // List of CI/CD variables.
+    $variables_list = $this->getSupportedVariables();
+    $variable_names = array_keys($variables_list);
+
+    $variable = NULL;
+    if ($input->getOption('gitlab-cicd-var')) {
+      $variable = $this->input->getOption('gitlab-cicd-var');
+      // If job provided by user is not valid.
+      if (!in_array($variable, $variable_names)) {
+        $this->io->error('Variable provided is not valid.');
+        return 1;
+      }
+    }
+
+    if (!$variable) {
+      $question = new ChoiceQuestion('Please select the Code Studio CI/CD variable you want to update.',
+        $variable_names,
+        $variable_names[0],
+      );
+      $variable = $this->io->askQuestion($question);
+    }
+
+    // Check if the variable already exists or not.
+    $all_project_variables = $this->gitLabClient->projects()->variables($project_id);
+    $is_variable_exist = FALSE;
+    foreach (array_values($all_project_variables) as $project_variable) {
+      if ($project_variable['key'] === $variable) {
+        $is_variable_exist = TRUE;
+        break;
+      }
+    }
+
+    $variable_value = '';
+    if ($this->input->getOption('gitlab-cicd-var-value')) {
+      // If user has provided the variable value as an option.
+      $variable_value = $this->input->getOption('gitlab-cicd-var-value');
+    }
+
+    // If variable value is not provided, ask user to enter it.
+    if (!$variable_value) {
+      $variable_val_question = new Question('Please enter the CI/CD variable value');
+      $variable_value = $this->io->askQuestion($variable_val_question);
```

Comment:
A lot of these variables should only accept either `true` or `false`. We probably need to define an array of acceptable values in `getSupportedVariables()`

Reply:
PR updated!